src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the proleptic-year and day-of-year fields.
+     * <p>
+     * This returns a {@code InternationalFixedDate} with the specified fields.
+     * The day must be valid for the year, otherwise an exception will be thrown.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param dayOfYear     the International fixed day-of-year, from 1 to 371
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the value of any field is out of range,
+     *                           or if the day-of-year is invalid for the year
+     */
+    static InternationalFixedDate ofYearDay (final int prolepticYear, final int dayOfYear) {
+        ChronoField.YEAR.checkValidValue (prolepticYear);
+        InternationalFixedChronology.DAY_OF_YEAR_RANGE.checkValidValue (dayOfYear, ChronoField.DAY_OF_YEAR);
+        boolean leap = InternationalFixedChronology.INSTANCE.isLeapYear (prolepticYear);
+
+        if (dayOfYear > (InternationalFixedChronology.DAYS_IN_YEAR + 1) && !leap) {
+            throw new DateTimeException ("Invalid date 'DayOfYear " + dayOfYear + "' as '" + prolepticYear + "' is not a leap year");
+        }
+
+        int month = ((dayOfYear - 1) / InternationalFixedChronology.DAYS_IN_MONTH) + 1;
+
+        // In leap years, the leap-month is shorter than the following month, so needs to be adjusted.
+        if (leap && month == InternationalFixedChronology.MONTHS_IN_YEAR && dayOfYear >= InternationalFixedChronology.DAYS_IN_YEAR - InternationalFixedChronology.DAYS_IN_MONTH + 1) {
+            month++;
+        }
+
+        // Subtract days-at-start-of-month from days in year
+        int dayOfMonth = dayOfYear - (month - 1) * InternationalFixedChronology.DAYS_IN_MONTH;
+
+        // Adjust for shorter inserted leap-month.
+        if (month == InternationalFixedChronology.MONTHS_IN_YEAR + 1) {
+            dayOfMonth += InternationalFixedChronology.DAYS_IN_MONTH;
+        }
+
+        return of (prolepticYear, month, dayOfMonth);
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the epoch-day.
+     *
+     * @param epochDay the epoch day to convert based on 1970-01-01 (ISO)
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the epoch-day is out of range
+     */
+    public static InternationalFixedDate ofEpochDay (final long epochDay) {
+        ChronoField.EPOCH_DAY.range ().checkValidValue (epochDay, ChronoField.EPOCH_DAY);
+        long zeroDay = epochDay + InternationalFixedChronology.DAYS_0000_TO_1970;
+
+        if (zeroDay < 0) {
+            throw new DateTimeException ("Negative epoch invalid: " + epochDay);
+        }
+
+        long yearEst = (400 * zeroDay) / DAYS_PER_CYCLE;
```

Comment:
This is labelled as an estimate, but I see no attempt (or comment about the lack) to true it up.  My gut is uneasy about the calculation I'm seeing, although I don't have anything to back that up.